### DISCUSSION TITLE
Custom models?

### DISCUSSION THREAD

**Discussion by Gamepro5** [nosoop/SM-TFCustomWeaponsX#47] [Q&A]
I run a website that facilitates the use of this plugin by generating the file for the user. I wanted to add an abstraction layer to custom models with custom weapons x, but I'm not sure how to go about it. Does custom weapons x require the models to be compressed into a VPK file? If not, would it be possible to reference a raw MDL file in the same directory as the config file for the attributes?

**Comment by nosoop** [comment]
Custom assets need to be cached and marked for download as bare models and textures.  At the moment this functionality isn't part of CWX and so another plugin would need to handle it, but it is one of the proposed use-cases for the "extdata" system added in #36.

0a86bfbfc7e6b4600ef8cb4c67929f836eba1db2 expands on this by allowing plugins to discover available custom items, so all the components for implementing this are in place, but no such plugin currently exists.  If time permits I may introduce a module and document it well enough for this.
